### ISSUE TITLE
Accessibility: Missing semantic markup for some elements of the plugin

### ISSUE THREAD

**Issue by mwreichel** [ndunand/moodle-mod_choicegroup#228]
Hi,

your activity is one of the most used plugins at the "Moodle an Hochschulen e.V." (https://moodle-an-hochschulen.de/) as well as in the German speaking Moodle community.
Therefore our working group on Accessibility did some tests on this, based on the German BITV 2.0 (a modified version of WCAG 2.1).

While the majority of the tests pasted, we found some minor issues regarding "Adaptability" (1.3) - Create content that can be presented in different ways (for example simpler layout) without losing information or structure. (https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=131#adaptable)

**Some elements don't have a semantic markup** (see: https://www.w3.org/WAI/WCAG21/Techniques/general/G115.html) and therefore it is not correct readable by assisting software, e.g. the button "Save my choice", the link "View x responses", the text "The result will...".
Please find these and some further examples of view.php marked in the attached screenshot.

![mod_groupchoice_9-1-3](https://github.com/user-attachments/assets/a008900d-b5e0-4d28-a62f-58d197a45cb6)

We tested this issue along the guidelines of BITV/ WCAG.

It would be glad if you could fix this bug. If this will be fixed, we could recommend your plugin as fully accessible within the German speaking community.

Many thanks in advance,
Martin
(for the working group "Accessibility")

**Comment by ndunand** [comment]
Dear Martin,

Thank for reporting this. Could you please elaborate as to which specific WCAG rules are not being followed, and to which elements you refer?

**Comment by mwreichel** [comment]
Dear @ndunand,

sorry for coming back to you so late.
In the meantime my colleagues re-tested this issue and there already was a partially enhancement. Now all of the listed groups in the table are marked correctly (please find attached an updated Screenshot of the current situation).

The other showed missing semantic markups are still there:
- the buttons "Show description" and "Save my choice"
- the link "View x responses"
- the paragraph with the information "The results will be visible after you have made your choice."

![Image](https://github.com/user-attachments/assets/b5e300c3-11b1-4aa6-a9a7-2acdc88678f8)

The relevant WCAG rule for this is:
- Technique G115: Using semantic elements to mark up structure (https://www.w3.org/WAI/WCAG21/Techniques/general/G115)

Thanks for working on this,
cheers

Martin
